src/components/ProductCard.tsx

```diff
@@ -81,6 +86,8 @@ export default function ProductCard({
             selectedColour={selectedColour}
             setSelectedColour={setSelectedColour}
             setCardImage={setCardImage}
+            setImageIndex={setImageIndex}
```

Comment:
Is this the entire purpose of dealing with the card color selection? I thought I gave you a solution for that already?